src/gui/advancedsettings.cpp

```diff
@@ -295,33 +288,35 @@ void AdvancedSettings::updateInterfaceAddressCombo()
 
     // Clear all items and reinsert them, default to all
     m_comboBoxInterfaceAddress.clear();
-    m_comboBoxInterfaceAddress.addItem(tr("All addresses"));
-    m_comboBoxInterfaceAddress.setCurrentIndex(0);
+    m_comboBoxInterfaceAddress.addItem(tr("All addresses"), {});
+    m_comboBoxInterfaceAddress.addItem(tr("All IPv4 addresses"), QLatin1String("0.0.0.0"));
+    m_comboBoxInterfaceAddress.addItem(tr("All IPv6 addresses"), QLatin1String("::"));
 
-    auto populateCombo = [this, &currentAddress](const QString &ip, const QAbstractSocket::NetworkLayerProtocol &protocol)
+    const auto populateCombo = [this, &currentAddress](const QHostAddress &addr)
     {
-        Q_ASSERT((protocol == QAbstractSocket::IPv4Protocol) || (protocol == QAbstractSocket::IPv6Protocol));
-        // Only take ipv4 for now?
-        if ((protocol != QAbstractSocket::IPv4Protocol) && (protocol != QAbstractSocket::IPv6Protocol))
-            return;
-        m_comboBoxInterfaceAddress.addItem(ip);
-        //Try to select the last added one
-        if (ip == currentAddress)
-            m_comboBoxInterfaceAddress.setCurrentIndex(m_comboBoxInterfaceAddress.count() - 1);
+        if (addr.protocol() == QAbstractSocket::IPv4Protocol) {
```

Comment:
Just a suggestion, it can be simplified to:
```c++
const QString str = (addr.protocol() == QAbstractSocket::IPv4Protocol)
    ? addr.toString()
    : Utils::Net::canonicalIPv6Addr(addr).toString();
m_comboBoxInterfaceAddress.addItem(str, str);
```
Assuming there won't be another IP version...

Reply:
In theory there are other values to the protocol enum: https://doc.qt.io/qt-5/qabstractsocket.html#NetworkLayerProtocol-enum
Practically it is unlikely to have one those other values.
In any case, I am leaning at leaving it they way it is now.